### ISSUE TITLE
discord.py death

### ISSUE THREAD

**Issue by MizaGBF** [MizaGBF/MizaBOT#12]
The bot will stay up until it stops working.  
If a good fork/alternative comes up before then, I'll port it over.  
If not, expect the bot to die.  
That's all.  
  
Context: https://gist.github.com/Rapptz/4a2f62751b9600a31a0d3c78100287f1

**Comment by MizaGBF** [comment]
Quit few forks spawned up since. I'll probably start experimenting with one of them in the near future.  
The end goal, for now, is the following:  
1. Port the bot to a good fork (ie. fix all breaking changes)  
2. Switch to slash commands (even if I don't like it)  
3. Remove all intents to avoid the server limit  

A few features might die (like the pinboard) but I'll do my best to work around.  
I don't know how many months it will take.

**Comment by MizaGBF** [comment]
Listing the changes made up to [v8.12c](https://github.com/MizaGBF/MizaBOT/commit/245ce491e664fbc58ec1ef8b2d2e4e3b91b3b75f):  
- save.json is now compressed to gzip (and vice versa) when sent/retrieved from gdrive (a simple tool is bundled with, in the tool folder, to do it on your own  
- `$youlead` has been upgraded with more infos  
- some special characters have been escaped/unescaped properly in `$findcrew` and `$findplayer`  
- updated the Procfile with the `-o` flag  
- updated `requirement.txt` with `uvloop` (for faster a faster asyncio event loop on linux) and the latest alpha of `discord.py` 2.0  
- added `$nm95`, `$xp` and `$poll`  
- removed `$logout` and `$reboot`
- all roll commands updated with a Tap button
- all link commands updated with Link buttons
- `$scratcher` and `$chest` updated to use interactions
- other unlisted bugfixes and smaller upgrades

**Comment by MizaGBF** [comment]
I started porting the bot to [disnake](https://github.com/DisnakeDev/disnake).  
I would say the work is about 90% done (not counting debugging/testing).  
  
A few things to note:  
* Every commands have been ported to either slash, user or message commands.  
As a result, the command list is pretty much unchanged (beside some old unused/not useful anymore commands I decided to remove, and some which got renamed).  
* The current invite link in the online help page is outdated. If you use it later when the invitations are open again, the bot will just not work (because it currently lacks the `application.commands` scope).  
* The save file version went up from 2 to 3. It consists mostly of minor changes (server prefixes, for example, have been removed).  
* Beside the discord.py to disnake changes, two libs will be removed from requirements.txt: cairosvg and leather. Using those two modules for a single function, which is exclusive to my own server, seemed like a waste, so I wrote my own replacement using Pillow.  
* the tool generate_help.py has been reworked to accommodate the new command types. It's also more detailed.  
  
The **v9.0** should go up in the next few weeks.  
I'm mostly waiting for a disnake patch to fix an issue with python 3.10, then I'll test/debug everything.

**Comment by MizaGBF** [comment]
I did some tests today:
* I wasn't aware of the 100 global slash commands limit, so I cleaned up the command list as a result (around 40 commands were removed). Some commands have been fused together. Other might come back later on (I'll pack as many as I can).  
* Debugging is mostly done, some errors might remain but I'll find them once the bot goes live.  

If it goes well, this new version will go live tomorrow.
